File: first-app/src/app/entity/student.ts
import {Clazz} from './clazz';

/**
 * 学生.
 */
export class Student {
  /**
   *  是否被选中
   */
  _checked = false;

  id: number;
  /**
   * 姓名.
   */
  name: string;
  /**
   * 学号.
   */
  number: string;
  /**
   * 手机号.
   */
  phone: string;
  /**
   * email.
   */
  email: string;
  /**
   * 班级.
   */
  clazz: Clazz;

  constructor(data = {} as
    {
      id?: number,
      name?: string,
      number?: string,
      phone?: string,
      email?: string,
      clazz?: Clazz
    }) {
    this.id = data.id as number;
    this.name = data.name as string;
    this.number = data.number as string;
    this.phone = data.phone as string;
    this.email = data.email as string;
    this.clazz = data.clazz as Clazz;
  }

  public onDeleteClick(): void {
    this._checked = !this._checked;
  }
}

File: angualrguide/src/main/java/club/yunzhi/api/angualrguide/entity/Student.java
package club.yunzhi.api.angualrguide.entity;

import com.fasterxml.jackson.annotation.JsonView;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@EntityListeners(AuditingEntityListener.class)
public class Student {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String name = "";

  @Column(nullable = false, unique = true)
  private String number;

  private String phone = "";

  private String email = "";

  @ManyToOne
  @JoinColumn(nullable = false)
  @JsonView(ClazzJsonView.class)
  private Clazz clazz;

  @CreatedDate
  private Timestamp createTime;

  public Student() {
  }

  public Student(Long id) {
    this.id = id;
  }

  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getNumber() {
    return number;
  }

  public void setNumber(String number) {
    this.number = number;
  }

  public String getPhone() {
    return phone;
  }

  public void setPhone(String phone) {
    this.phone = phone;
  }

  public String getEmail() {
    return email;
  }

  public void setEmail(String email) {
    this.email = email;
  }

  public Clazz getClazz() {
    return clazz;
  }

  public void setClazz(Clazz clazz) {
    this.clazz = clazz;
  }

  public Timestamp getCreateTime() {
    return createTime;
  }

  public void setCreateTime(Timestamp createTime) {
    this.createTime = createTime;
  }

  public interface ClazzJsonView {
  }
}
